### PR TITLE
Hide OT logic inside the implementation file

### PR DESCRIPTION
**Pull request by Lucashuang0802**
***Goal***: 
Try to reduce the confusion from the OpenTok platform SDK

***Approach***:
- hide all the OpenTok logic into the implement files in `oneToOneCommunicator`
- expose developer friendly APIs in `oneToOneCommunicator`

***Benefit***:
- developers can grab the `oneToOneCommunicator` directly and use it
- convenient for us to integrate into other sample apps

@marinaserranomontes @aradiquez, please review and consider give :+1:. We can follow this guide in order to allow contribution.

### CODE REVIEW

File: one-to-one-sample-app/iOS/SampleApp/MainViewController.m [hunk 1]
```diff
@@ -22,7 +22,6 @@ - (void)viewDidLoad {
 - (IBAction)publisherCallButtonPressed:(UIButton *)sender {
     if (self.oneToOneCommunicator.isCallEnabled) {
         [self.mainView callHolderDisconnected];
-        self.oneToOneCommunicator.isCallEnabled = NO;
```
Comment (aradiquez) [inline one-to-one-sample-app/iOS/SampleApp/MainViewController.m]: Nice, you include this code inside the connect and disconnect methods, really cool

**Comment by aradiquez** [comment]
:+1:

File: one-to-one-sample-app/iOS/SampleApp/MainViewController.m [hunk 2]
```diff
@@ -85,18 +83,18 @@ - (void)handleCommunicationSignal:(OneToOneCommunicationSignal)signal {
...
         case OneToOneCommunicationSignalSubscriberVideoEnabled:{
             [self.mainView hideErrorMessageLabel];
-            [self.mainView addSubscribeView:self.oneToOneCommunicator.subscriber.view];
+            [self.mainView addSubscribeView:self.oneToOneCommunicator.subscriberView];
             break;
         }
         case OneToOneCommunicationSignalSubscriberVideoDisableWarning:{
             [self.mainView addPlaceHolderToSubscriberView];
-            self.oneToOneCommunicator.subscriber.subscribeToVideo = NO;
+            self.oneToOneCommunicator.subscribeToVideo = NO;
```
Comment (aradiquez) [inline one-to-one-sample-app/iOS/SampleApp/MainViewController.m]: pretty cool to know we can use setters like this!!

File: one-to-one-sample-app/iOS/SampleApp/MainViewController.m (hunk 1 above)
Comment (Lucashuang0802) [inline one-to-one-sample-app/iOS/SampleApp/MainViewController.m]: I think about this again and the responsibility of updating this bool value should belong to model for sure.

**Comment by Lucashuang0802** [comment]
Close because it has minor front-end bug, @aradiquez is going to work on it and reopen.

**Pull request closed without merging**